webdriver/tests/bidi/network/fail_request/invalid.py

```diff
@@ -18,18 +25,20 @@ async def test_params_request_invalid_value(bidi_session, value):
         await bidi_session.network.fail_request(request=value)
 
 
-async def test_params_request_no_such_request(bidi_session, setup_network_test,
-                                              wait_for_event, wait_for_future_safe,
-                                              fetch, url):
-    await setup_network_test(events=[
-        RESPONSE_COMPLETED_EVENT,
-    ])
+async def test_params_request_no_such_request(
+    bidi_session, setup_network_test, wait_for_event, fetch, url
+):
+    await setup_network_test(
+        events=[
+            RESPONSE_COMPLETED_EVENT,
+        ]
+    )
     on_response_completed = wait_for_event(RESPONSE_COMPLETED_EVENT)
 
     text_url = url(PAGE_EMPTY_TEXT)
     await fetch(text_url)
 
-    response_completed_event = await wait_for_future_safe(on_response_completed)
+    response_completed_event = await on_response_completed
```

Comment:
Could this change maybe be the reason why the [wpt.fyi results show a timeout](https://wpt.fyi/results/webdriver/tests/bidi/network/fail_request/invalid.py?diff&filter=ADC&run_id=5094192866394112&run_id=5148044844072960) for this test file now with Chrome?

Reply:
No, the job seems to timeout on the invalid phase test.
The spec is quite clear about the fact that this should be an error:

> If phase is "authRequired", then return [error](https://w3c.github.io/webdriver/#dfn-errors) with [error code](https://w3c.github.io/webdriver/#dfn-error-code) [invalid argument](https://w3c.github.io/webdriver/#dfn-invalid-argument).

But even if they don't implement the phase check this shouldn't timeout.

Chrome also seems to timeout on various other tests: 
- continue_with_auth/invalid.py::test_params_request_invalid_phase[responseStarted]
- continue_response/invalid.py::test_params_reason_phrase_invalid_type[False] 

All those tests have in common that they need to block a request in another phase than "beforeRequest". I remember that during the initial review of this patch, I requested several times to change the phase of tests from "beforeRequest" to the appropriate phase but this was not addressed, so it's possible that chrome doesn't support blocking the request in phases after "beforeRequest". 

Looking at wpt.fyi for other intercept test, my guess is that chrome is always adding intercepts in the beforeRequest phase, even if you are specifying another phase. This would be enough to make the `setup_blocked_request` helper timeout. 

cc @sadym-chromium can you confirm or tell me where the code handling intercepts is located so that I can check?

Reply:
Unfortunately, for now our network interception implementation is racy and sometimes hangs forever. We are working on fixing it.

Reply:
Wait. This test does not set interceptions, so it should work just fine. Let me check.

Reply:
Sorry it's a bit misleading :) 

The question from @whimboo was added for this test (`test_params_request_no_such_request`), but the actual timeout happens in `test_params_request_invalid_phase` (right above this one) which indirectly uses `add_intercept` via `setup_blocked_request`